src/Classes/ServiceAPI/MyRadio_Track.php

```diff
@@ -770,10 +770,21 @@ public function getSimilar() {
             }
             foreach ($data['similartracks']['track'] as $r) {
                 if ($r['match'] >= 0.25) {
+                    //Try to find an exact match
                     $c = self::findByOptions(['title' => $r['name'],
                                 'artist' => $r['artist']['name'],
                                 'limit' => 1,
-                                'digitised' => true]);
+                                'digitised' => true
+                                'precise' => true]);
+                    //Try to find a not-so-exact match
+                    if (!empty($c)) {
+                        $c = self::findByOptions(['title' => $r['name'],
+                                'artist' => $r['artist']['name'],
+                                'limit' => 1,
+                                'digitised' => true
+                                'precise' => false]);
+                    }
+                    //Still nothing
```

Comment:
This comment is not true, the next condition only runs if there is something (it adds the found track to the array lastfm_similar)
Better comment would be something like: Add the track to the similar list if we have it